rticonnextdds_connector.go

```diff
@@ -201,49 +201,36 @@ func (connector *Connector) Delete() (err error) {
 }
 
 // GetOutput returns an output object
-func (connector *Connector) GetOutput(outputName string) (output *Output, err error) {
+func (connector *Connector) GetOutput(outputName string) (*Output, error) {
 	if connector == nil {
-		err = errors.New("Connector is null")
-		return nil, err
+		return nil, errors.New("connector is null")
 	}
 
-	output, err = newOutput(connector, outputName)
-	if err != nil {
-		return nil, err
-	}
-	return output, nil
+	return newOutput(connector, outputName)
 }
 
 // GetInput returns an input object
-func (connector *Connector) GetInput(inputName string) (input *Input, err error) {
+func (connector *Connector) GetInput(inputName string) (*Input, error) {
 	if connector == nil {
-		err = errors.New("Connector is null")
-		return nil, err
+		return nil, errors.New("connector is null")
 	}
 
-	input, err = newInput(connector, inputName)
-	if err != nil {
-		return nil, err
-	}
-	return input, nil
+	return newInput(connector, inputName)
 }
 
 // Wait is a function to block until data is available on an input
-func (connector *Connector) Wait(timeoutMs int) (err error) {
+func (connector *Connector) Wait(timeoutMs int) error {
 	if connector == nil {
-		err = errors.New("Connector is null")
-		return err
+		return errors.New("connector is null")
 	}
-
-	retcode := int(C.RTIDDSConnector_wait(unsafe.Pointer(connector.native), (C.int)(timeoutMs)))
-	if retcode == 10 /* DDS_RETCODE_TIMEOUT */ {
-		err = errors.New("Timeout")
-		return err
-	} else if retcode != 0 /* DDS_RETCODE_OK */ {
-		err = errors.New("RTIDDSConnector_wait error")
-		return err
+	switch int(C.RTIDDSConnector_wait(unsafe.Pointer(connector.native), C.int(timeoutMs))) {
```

Comment:
@metarsit Can we also add the following?

case DDSRetCodeNoData:
  return errors.New("no data")

Reply:
Added